From e3f6ffeab976bb070f502478669c136307540aa6 Mon Sep 17 00:00:00 2001
Message-ID: <e3f6ffeab976bb070f502478669c136307540aa6.1753541716.git.lorenzo@kernel.org>
In-Reply-To: <f797f67cd7917f060f4296742e2fa8a9f4bd0f39.1753541716.git.lorenzo@kernel.org>
References: <f797f67cd7917f060f4296742e2fa8a9f4bd0f39.1753541716.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Fri, 25 Jul 2025 12:38:26 +0200
Subject: [PATCH 3/5] wifi: mt76: Add the capability to set TX token start ID

This is a preliminary patch to enable traffic forward offloading via the
Airoha NPU module available on the Airoha EN7581 SoC.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 mt76.h | 1 +
 tx.c   | 6 ++++--
 2 files changed, 5 insertions(+), 2 deletions(-)

--- a/mt76.h
+++ b/mt76.h
@@ -1122,6 +1122,7 @@ struct mt76_dev {
 	struct idr token;
 	u16 wed_token_count;
 	u16 token_count;
+	u16 token_start;
 	u16 token_size;
 	u16 token_threshold;
 
--- a/tx.c
+++ b/tx.c
@@ -862,7 +862,7 @@ EXPORT_SYMBOL_GPL(__mt76_set_tx_blocked)
 int mt76_token_consume(struct mt76_dev *dev, struct mt76_txwi_cache **ptxwi,
 		       u8 phy_idx)
 {
-	int token = -EINVAL, start = 0;
+	int token = -EINVAL, start = dev->token_start;
 	struct mt76_phy *phy = mt76_dev_phy(dev, phy_idx);
 	#ifdef CONFIG_NET_MEDIATEK_SOC_WED	
 	if (mtk_wed_device_active(&dev->mmio.wed))
